### PR TITLE
add checkpoint key in config

### PR DESCRIPTION
**Pull request by benjijamorris**
## What does this PR do?
- add config section called "checkpoint" that includes ckpt_path, whether to load weights only from the ckpt_path, and whether to load state dict strictly

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?
- [x] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: cyto_dl/train.py [hunk 1]
```diff
@@ -96,23 +96,24 @@ def train(cfg: DictConfig, data=None) -> Tuple[dict, dict]:
...
+        load_params = cfg.get("checkpoint")
+        if load_params.get("weights_only"):
+            assert load_params.get(
                 "ckpt_path"
             ), "ckpt_path must be provided to with argument weights_only=True"
             # load model from state dict to get around trainer.max_epochs limit, useful for resuming model training from existing weights
-            state_dict = torch.load(cfg["ckpt_path"])["state_dict"]
-            model.load_state_dict(state_dict)
-            cfg["ckpt_path"] = None
+            state_dict = torch.load(load_params["ckpt_path"])["state_dict"]
```
Comment (ritvikvasan) [inline cyto_dl/train.py:105]: missing ) here

File: cyto_dl/train.py [hunk 2]
```diff
@@ -96,23 +96,24 @@ def train(cfg: DictConfig, data=None) -> Tuple[dict, dict]:
     if cfg.get("train"):
         log.info("Starting training!")
 
-        if cfg.get("weights_only"):
-            assert cfg.get(
+        load_params = cfg.get("checkpoint")
+        if load_params.get("weights_only"):
```
Comment (ritvikvasan) [inline cyto_dl/train.py:100]: what's the difference between loading the state dict to the model and setting ckpt_path to None vs just passing in ckpt_path to the trainer?

**Review by ritvikvasan — APPROVED** [review]
Looks good!

File: cyto_dl/train.py (hunk 2 above)
Comment (benjijamorris) [reply cyto_dl/train.py:100]: passing the ckpt_path to the trainer also loads non-state dict optimization metadata (and always does a strict load). For the plugin we want to e.g. let people finetune megaseg, so we want to only load the weights to let people continue training past the original models `trainer.max_epochs`

Comment (ritvikvasan) [reply cyto_dl/train.py:100]: oh gotcha, maybe just add exactly this as a comment there?

File: cyto_dl/train.py (hunk 1 above)
Comment (ritvikvasan) [reply cyto_dl/train.py:105]: nvm my bad

**Review by yrkim98 — APPROVED** [review]
thank you! ping me when this is released and ill run some more tests

**Comment by yrkim98** [comment]
one question- if we define a `ckpt_path`, and set `weights_only=True`, will a new `train_config.yaml` be saved to the `output_dir`? or is it on the plugin to make a copy and save it?

**Comment by benjijamorris** [comment]
This shouldn't change the saving behavior - if `print_config=True` in the config (True by default), the config will still be saved out

**Pull request merged**